crypto/o_time.c

```diff
@@ -47,6 +47,33 @@ typedef struct _ile3 {          /* Copied from ILEDEF.H for Alpha   */
 # endif   /* alpha || ia64    */
 #endif    /* OPENSSL_SYS_VMS  */
 
+time_t OPENSSL_timegm(struct tm *tm)
+{
+#if defined(__USE_MISC)
+    return timegm(tm);
+#else
+    /* Figures it out without mucking the environment */
+    time_t currtime, utc, local;
+    struct tm *timeinfo;
+    int offset;
+
+    time(&currtime);
+    timeinfo = gmtime(&currtime);
+    utc = mktime(timeinfo);
+
+    timeinfo = localtime(&currtime);
+    local = mktime(timeinfo);
+
+    offset = (int)difftime(utc, local);
+
+    /* Adjust for DST */
+    if (timeinfo->tm_isdst)
+        offset -= 3600;
+
+    return mktime(tm) - offset;
```

Comment:
Really wondering if there is a better/standard/non-system-specific way to get the timezone offset, or at least emulate the non-standard timegm(). This is ugly, but it appears to work.